### ISSUE TITLE
Cannot clone repository

### ISSUE THREAD

**Issue by rudolfbyker** [aws-amplify/amplify-hosting#3162]
### Before opening, please confirm:

- [X] I have checked to see if my question is addressed in the [FAQ](https://github.com/aws-amplify/amplify-hosting/blob/master/FAQ.md).
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-hosting/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-hosting/blob/master/CONTRIBUTING.md).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### App Id

d2ypd951wv2365

### AWS Region

eu-west-2

### Amplify Hosting feature

Frontend builds, Git providers

### Describe the bug

My issue seems to be exactly the same as https://github.com/aws-amplify/amplify-hosting/issues/3079 but that one is closed, and it is not solved for me.

### Expected behavior

Amplify can clone my private repo because it is authorized via a Github App.

### Reproduction steps

1. Trigger a front end build
2. Clone fails. Log output is below.

### Build Settings

```yaml
version: 1
frontend:
  phases:
    preBuild:
      commands:
        - yarn install
    build:
      commands:
        - yarn run build
  artifacts:
    baseDirectory: /dist/
    files:
      - '**/*'
  cache:
    paths:
      - node_modules/**/*
```

### Log output

<details>

```
# Put your logs below this line

2022-12-01T13:36:17.639Z [INFO]: Git SSH Key acquired
2022-12-01T13:36:17.707Z [INFO]: # Cloning repository: git@github.com:AutoActuary/autoactuary.com.git
2022-12-01T13:36:20.181Z [INFO]: Agent pid 142
2022-12-01T13:36:20.183Z [INFO]: Identity added: /root/.ssh/git_rsa (/root/.ssh/git_rsa)
                                 Cloning into 'autoactuary.com'...
                                 Warning: Permanently added the ECDSA host key for IP address '140.82.121.4' to the list of known hosts.
                                 Permission denied (publickey).
                                 fatal: Could not read from remote repository.
                                 Please make sure you have the correct access rights
                                 and the repository exists.
2022-12-01T13:36:20.183Z [ERROR]: !!! CustomerError: Unable to clone repository due to user error code: 128
2022-12-01T13:36:20.183Z [INFO]: # Starting environment caching...
2022-12-01T13:36:20.184Z [INFO]: # Environment caching completed
Terminating logging...
```

</details>

### Additional information

What I have tried:

- Re-connect repository
- Delete the app on Github and re-connect respository
- Trigger via pushing a new commit
- Trigger via command line like `aws amplify start-job --app-id d2ypd951wv2365 --branch-name deploy-prod --job-type RELEASE`

What I cannot afford to try:

- Deleting and re-creating the app (i.e. I can't have downtime)

**Comment by hloriana** [comment]
Hi @rudolfbyker 👋🏽  We have not been able to reproduce this behavior as of yet but we are working to understand why cloning is failing for your repository. Does your repository belong to a GitHub organization by any chance? If it is, do you have permissions to grant Amplify access to that repository?

**Comment by rudolfbyker** [comment]
> Does your repository belong to a GitHub organization by any chance? If it is, do you have permissions to grant Amplify access to that repository?

Yes, and yes. It has worked before, with the same repository and the same organization and the same amplify project. I just have not deployed in a long time. When I made a few changes to the website's text, and committed them to github, the deploy failed.

**Comment by hloriana** [comment]
Hi @rudolfbyker 👋🏽 It looks like your last successful deployment was in Feb 2021. During that time, Amplify used OAuth to receive access to repositories. We now use a GitHub app to grant permissions. Just to confirm, have you migrated from using OAuth to using the Amplify GitHub app? 

Here are the steps to do so: https://docs.aws.amazon.com/amplify/latest/userguide/setting-up-GitHub-access.html#migrating-to-github-app-auth

**Comment by rudolfbyker** [comment]
👋 Thanks for your time.

> have you migrated from using OAuth to using the Amplify GitHub app?

Yes. When I do this, it says that it was successful, and I can see the App registered on Github, but the message on the console asking me to migrate never goes away. I tried this multiple times.

I also tried removing anything that I could see related to OAuth from the GitHub side, but to no avail.

**Comment by elquimeras** [comment]
I have the same problem from the past week, I tried everything but notting happens... We have the github app installed, and the amplify are working before, but from the past week the compilation step says that cannot connect to github the fetch the code: 


```
2022-12-12T15:15:23.571Z [INFO]: Git SSH Key acquired
2022-12-12T15:15:23.671Z [INFO]: # Cloning repository: git@github.com:myrepo.git
2022-12-12T15:15:26.188Z [INFO]: Agent pid 143
2022-12-12T15:15:26.189Z [INFO]: Identity added: /root/.ssh/git_rsa (/root/.ssh/git_rsa)
                                 Cloning into 'myrepo'...
                                 Warning: Permanently added the ECDSA host key for IP address '192.30.255.113' to the list of known hosts.
                                 Permission denied (publickey).
                                 fatal: Could not read from remote repository.
                                 Please make sure you have the correct access rights
                                 and the repository exists.
2022-12-12T15:15:26.190Z [ERROR]: !!! CustomerError: Unable to clone repository due to user error code: 128
```

**Comment by elquimeras** [comment]
@rudolfbyker @manalab17 
I fixed the issue uninstalling the Amplify github app from my repo, and then reinstalling it, with https://github.com/apps/aws-amplify-us-west-2/installations/new (remember to change the región if not is us-west-2 yours)

**Comment by rudolfbyker** [comment]
> @rudolfbyker @manalab17 I fixed the issue uninstalling the Amplify github app from my repo, and then reinstalling it, with https://github.com/apps/aws-amplify-us-west-2/installations/new (remember to change the región if not is us-west-2 yours)

I already tried that.

**Comment by hloriana** [comment]
Hi @rudolfbyker apologies for the delay in resolving this. Since the common workarounds are not working for you and we cannot reproduce the behavior would you be available for a call so we can do a deep dive? Please reach out to:
aws-amplify-customer@amazon.com with your availability.

**Comment by hloriana** [comment]
Thanks for taking the time to meet earlier today to go over this issue. I'll close this out since we were able to get the application to successfully build. Feel free to reach out if you need assistance with anything else Amplify related.
